projects/core/src/routing/event/routing-event.builder.ts

```diff
@@ -0,0 +1,164 @@
+    return this.getNavigatedEvent().pipe(
+      map((state) => createFrom(PageVisited, state))
+    );
+  }
+
+  protected buildHomePageVisitedEvent(): Observable<HomePageVisited> {
+    return this.getCurrentPageContextFor('home').pipe(
+      map((pageContext) => createFrom(HomePageVisited, pageContext))
+    );
+  }
+
+  protected buildProductDetailsPageVisitedEvent(): Observable<
+    ProductDetailsPageVisited
+  > {
+    return this.getCurrentPageContextFor('product').pipe(
+      map((context) => context.id),
+      switchMap((productId) => {
+        return this.productService.get(productId).pipe(
+          filter(Boolean),
+          map((product) => {
+            return createFrom(ProductDetailsPageVisited, product);
+          })
+        );
+      })
+    );
+  }
+
+  protected buildCartVisitedEvent(): Observable<CartPageVisited> {
+    return this.getCurrentPageContextFor('cart').pipe(
+      map((pageContext) => createFrom(CartPageVisited, pageContext))
+    );
+  }
+
+  protected orderConfirmationPageVisitedEvent(): Observable<
+    OrderConfirmationPageVisited
+  > {
+    return this.getCurrentPageContextFor('orderConfirmation').pipe(
+      map((pageContext) =>
+        createFrom(OrderConfirmationPageVisited, pageContext)
+      )
+    );
+  }
+
+  //FIXME: see https://github.com/SAP/spartacus/issues/7991
+  protected buildCategoryPageVisitedEvent(): Observable<CategoryPageVisited> {
+    return this.productSearchService.getResults().pipe(
+      filter((searchResults) => {
+        return (
+          searchResults.breadcrumbs && searchResults.breadcrumbs.length > 0
+        );
+      }),
+      withLatestFrom(this.getCurrentPageContextFor('category')),
+      filter(
+        ([searchResults, _pageContext]) =>
+          searchResults.breadcrumbs && searchResults.breadcrumbs.length > 0
+      ),
+      map(([searchResults, pageContext]) => ({
+        categoryCode: pageContext.id,
+        categoryName: searchResults.breadcrumbs[0].facetValueName,
+      })),
+      map((categoryPage) => createFrom(CategoryPageVisited, categoryPage))
+    );
+  }
+
+  protected searchResultPageVisited(): Observable<SearchPageVisited> {
+    return this.productSearchService.getResults().pipe(
+      filter((searchResults) => Boolean(searchResults.breadcrumbs)),
+      withLatestFrom(this.getCurrentPageContextFor('search')),
+      map(([productSearchPage, _pageContext]) => ({
+        searchTerm: productSearchPage.freeTextSearch,
+        numberOfResults: productSearchPage.pagination.totalResults,
+      })),
+      map((searchResults) => createFrom(SearchPageVisited, searchResults))
+    );
+  }
+
+  private getNavigatedEvent(): Observable<ActivatedRouterStateSnapshot> {
+    return this.actions.pipe(
+      ofType<RouterNavigatedAction>(ROUTER_NAVIGATED),
+      map(
+        (event) =>
+          (event.payload.routerState as unknown) as ActivatedRouterStateSnapshot
+      )
+    );
+  }
+
+  private getCurrentPageContextFor(
```

Comment:
Medium: By the name I don't know what this method does. Gets current page context? Or gets page context of the given semantic route? A bit confusing to me. 
Can we /** document */ what problem this method solves?

Reply:
It appears this has been moved to another PR